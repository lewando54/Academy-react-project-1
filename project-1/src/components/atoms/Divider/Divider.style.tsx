import styled from 'styled-components'

export const StyledHr = styled.hr`
    border-top: 2px solid #e5e7eb;
    width: 100%;
`

export const StyledDiv = styled.div`
    display: flex;
    width: 100%;
    align-items: center;
`

export const StyledP = styled.p`
    padding: 0.25rem/* 4px */;
    border: 2px solid rgb(156 163 175);
    opacity: 0.3;
    border-radius: 0.375rem;
    font-size: 0.875rem/* 14px */;
    line-height: 1.25rem/* 20px */;
    margin: 0;
`
