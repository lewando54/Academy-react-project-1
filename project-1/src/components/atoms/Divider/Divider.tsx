import React from 'react'
import { StyledDiv, StyledHr, StyledP } from './Divider.style'

interface IDividerProps {
  text?: string
  testId: string
}

function Divider ({ text = '', testId }: IDividerProps): React.ReactElement {
  return (
        <StyledDiv data-testid={testId}>
            <StyledHr/>
            { (text.length > 0) ? <StyledP>{text}</StyledP> : null}
            <StyledHr/>
        </StyledDiv>
  )
}

export default Divider
