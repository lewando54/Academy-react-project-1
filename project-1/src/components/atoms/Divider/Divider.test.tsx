import React from 'react'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import Divider from './Divider'

describe('Divider', () => {
  it('renders correctly', () => {
    const { container } = render(<Divider />)
    expect(container.firstChild).toBeInTheDocument()
  })

  it('renders with optional text', () => {
    const { container } = render(<Divider text='Hello World!' />)
    expect(container.querySelector('p')).toHaveTextContent('Hello World!')
  })

  it('does not render text when text prop is empty', () => {
    const { container } = render(<Divider />)
    expect(container.querySelector('p')).toBeNull()
  })
})
