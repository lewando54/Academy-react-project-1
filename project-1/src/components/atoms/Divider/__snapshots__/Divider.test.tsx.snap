// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should not render text by default 1`] = `
<div
  className="sc-gswNZR czINuU"
  data-testid="divider"
>
  <hr
    className="sc-bcXHqe dtRlJI"
  />
  <hr
    className="sc-bcXHqe dtRlJI"
  />
</div>
`;

exports[`Divider should render correctly 1`] = `
<div
  className="sc-gswNZR czINuU"
  data-testid="divider"
>
  <hr
    className="sc-bcXHqe dtRlJI"
  />
  <hr
    className="sc-bcXHqe dtRlJI"
  />
</div>
`;

exports[`Divider should render with optional text 1`] = `
<div
  className="sc-gswNZR czINuU"
  data-testid="divider"
>
  <hr
    className="sc-bcXHqe dtRlJI"
  />
  <p
    className="sc-dkrFOg hEXefI"
  >
    Hello World!
  </p>
  <hr
    className="sc-bcXHqe dtRlJI"
  />
</div>
`;
